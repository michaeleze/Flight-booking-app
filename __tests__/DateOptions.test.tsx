import { act, cleanup, fireEvent, render, RenderResult, screen } from '@testing-library/react';
import DateOption from '@/components/DateOption';

const LABEL = "Departure date";
const DEPATURE_DATE = '01.10.2020';
const handleChangeDepartureDate = jest.fn();

describe('DateOption', () => {
  let component;

  beforeEach(() => {
    act(() => {
      component = render(
        <DateOption
          label={LABEL}
          handleSelectDate={handleChangeDepartureDate}
        />);
    });
  });

  afterEach(cleanup);

  it('should renders component', async () => {
    const { getByTestId } = await component;
    const label = getByTestId(/date-input--label/i);

    expect(label).toBeInTheDocument();
  });

  it('should fire onChange',async  () => {
    const { getByTestId } = await component;

    fireEvent.change(getByTestId('date-input'), { target: { value: DEPATURE_DATE } })

    expect(handleChangeDepartureDate).toHaveBeenCalledOnce();
  });

  it('should fire onChange', async () => {
    const { getByTestId } = await component;

    fireEvent.change(getByTestId('date-input'), { target: { value: DEPATURE_DATE } })

    expect(component).toHaveValue(DEPATURE_DATE);
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
