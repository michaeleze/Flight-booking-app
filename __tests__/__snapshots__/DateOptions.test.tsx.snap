// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateOption should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="group relative z-0 mb-6 w-1/4"
      >
        <div
          class="group relative z-0 mb-6 w-full"
        >
          <input
            class="peer block w-full appearance-none border-0 border-b-2 border-gray-100 bg-transparent py-2.5 px-0 text-sm text-gray-900 focus:border-blue-600 focus:outline-none focus:ring-0 dark:border-gray-600 dark:text-white dark:focus:border-blue-500"
            data-testid="date-input"
            name="floating_email"
            required=""
            type="date"
            value=""
          />
          <label
            class="absolute top-3 -z-10 origin-[0] -translate-y-6 scale-75 transform text-sm text-gray-500 duration-300 peer-placeholder-shown:translate-y-0 peer-placeholder-shown:scale-100 peer-focus:left-0 peer-focus:-translate-y-6 peer-focus:scale-75 peer-focus:font-medium peer-focus:text-blue-600 dark:text-gray-400 peer-focus:dark:text-blue-500"
            data-testid="date-input--label"
            for="floating_email"
          >
            Departure date
          </label>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="group relative z-0 mb-6 w-1/4"
    >
      <div
        class="group relative z-0 mb-6 w-full"
      >
        <input
          class="peer block w-full appearance-none border-0 border-b-2 border-gray-100 bg-transparent py-2.5 px-0 text-sm text-gray-900 focus:border-blue-600 focus:outline-none focus:ring-0 dark:border-gray-600 dark:text-white dark:focus:border-blue-500"
          data-testid="date-input"
          name="floating_email"
          required=""
          type="date"
          value=""
        />
        <label
          class="absolute top-3 -z-10 origin-[0] -translate-y-6 scale-75 transform text-sm text-gray-500 duration-300 peer-placeholder-shown:translate-y-0 peer-placeholder-shown:scale-100 peer-focus:left-0 peer-focus:-translate-y-6 peer-focus:scale-75 peer-focus:font-medium peer-focus:text-blue-600 dark:text-gray-400 peer-focus:dark:text-blue-500"
          data-testid="date-input--label"
          for="floating_email"
        >
          Departure date
        </label>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
